File: ios_app/CatchMeNearby/Core/Firebase/FirebaseService.swift
import Foundation
import FirebaseAuth
import FirebaseFirestore
import FirebaseStorage

class FirebaseService {
    static let shared = FirebaseService()

    let auth = Auth.auth()
    let firestore = Firestore.firestore()
    let storage = Storage.storage()

    private init() {}

    // Placeholder for common Firebase operations
}

File: android_app/app/src/main/java/com/catchmenearby/core/firebase/FirebaseService.java
package com.catchmenearby.core.firebase;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.storage.FirebaseStorage;

public class FirebaseService {
    private static FirebaseService instance;

    public final FirebaseAuth auth;
    public final FirebaseFirestore firestore;
    public final FirebaseStorage storage;

    private FirebaseService() {
        auth = FirebaseAuth.getInstance();
        firestore = FirebaseFirestore.getInstance();
        storage = FirebaseStorage.getInstance();
    }

    public static synchronized FirebaseService getInstance() {
        if (instance == null) {
            instance = new FirebaseService();
        }
        return instance;
    }
    // Placeholder for common Firebase operations
}
